### DISCUSSION TITLE
Allocate more resources to Quartz service

### DISCUSSION THREAD

**Discussion by brent-taz** [quartznet/quartznet#2586] [Q&A]
We have a Windows service written in .NET Framework 4.7 that uses Quartz 3.8 to schedule long running jobs.
Been running for a few years and very happy with it, however, some jobs take several hours to complete.

Looking at Task Manager, the windows service never gets more than maybe 3% CPU utilization.
I thought it might be a Windows service limitation, but I've tried running it as a stand alone application with the same results.

Can anyone point me in the right direction on how to get the app to use more resources?
I'm sure if we can allocate more, these jobs would complete much faster.

Thanks,
Brent

**Comment by lahma** [comment]
I would start by running the code as standalone without Quartz to see if it is indeed Quartz that is slowing you down.

**Comment by tomap** [comment]
I would suppose that your service (CPU/RAM) is not the bottleneck. 
Have you checked IO (disk access) ?

Using some kind of tracing would help to determine what dependencies are called (DB, network,...) 

Some services provide that of the shelf (Elastic APM, Azure App Insight, Sentry, Datadog, ...) providing you can install then & use them in your context
